v19.8.3 - Expected an assignment or function call and instead saw an expression

### Current Behavior

I encountered the following ESLint error when running the nx run-many -t lint command for a newly created workspace with plugin

Expected an assignment or function call and instead saw an expression. eslint@typescript-eslint/no-unused-expressions

### Expected Behavior

The linting step should not fail with the @typescript-eslint/no-unused-expressions error

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create a new empty workspace
2. Install @nx/plugin
3. Create a new plugin with e2e
4. Run lint


### Nx Report

```shell
❯ nx report              

 NX   Report complete - copy this into the issue template

Node           : 22.9.0
OS             : darwin-x64
Native Target  : x86_64-macos
npm            : 10.8.3

nx (global)        : 19.8.3
nx                 : 19.8.3
@nx/js             : 19.8.3
@nx/jest           : 19.8.3
@nx/linter         : 19.8.3
@nx/eslint         : 19.8.3
@nx/workspace      : 19.8.3
@nx/devkit         : 19.8.3
@nx/eslint-plugin  : 19.8.3
@nx/plugin         : 19.8.3
@nrwl/tao          : 19.8.3
typescript         : 5.5.4
```


### Failure Logs

```shell
❯ nx run-many -t lint

   ✔  nx run my-plugin:lint  [existing outputs match the cache, left as is]

——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
   ✖  nx run my-plugin-e2e:lint
      Linting "my-plugin-e2e"...
      
      /Users/gperdomor/Downloads/nx-tools-v7/plugins/my-plugin-e2e/project.json
        1:1  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
      
      /Users/gperdomor/Downloads/nx-tools-v7/plugins/my-plugin-e2e/tsconfig.json
        1:1  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
      
      /Users/gperdomor/Downloads/nx-tools-v7/plugins/my-plugin-e2e/tsconfig.spec.json
        1:1  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
      
      ✖ 3 problems (3 errors, 0 warnings)
      
      ✖ 3 problems (3 errors, 0 warnings)
      
      

——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Ran target lint for 2 projects (2s)

   ✔  1/2 succeeded [1 read from cache]

   ✖  1/2 targets failed, including the following:

      - nx run my-plugin-e2e:lint

View structured, searchable error logs at https://nx.app/runs/zUZJzouWU1
```


### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Related to #28069

@jaysoo I see this issue with react workspace too 


> eslint .


/Users/s.yadav/react-india/libs/my-awesome-lib/package.json
  1:1  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions

I am on version Nx 19.8.4

--- a/packages/eslint-plugin/src/flat-configs/react-base.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-base.ts
@@ -18,11 +18,16 @@ import tseslint from 'typescript-eslint';
  * https://github.com/facebook/create-react-app/blob/567f36c9235f1e1fd4a76dc6d1ae00be754ca047/packages/eslint-config-react-app/index.js
  */
 export default tseslint.config({
-  // Files need to be specified or else typescript-eslint rules will be
-  // applied to non-TS files. For example, libs add rules to *.json
-  // files, and TS rules should not apply to them.
-  // See: https://github.com/nrwl/nx/issues/28245
-  files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
+  files: [
+    '**/*.ts',
+    '**/*.cts',
+    '**/*.mts',
+    '**/*.tsx',
+    '**/*.js',
+    '**/*.cjs',
+    '**/*.mjs',
+    '**/*.jsx',
+  ],
   plugins: { import: importPlugin },
   languageOptions: {
     globals: {

--- a/packages/eslint-plugin/src/flat-configs/react-base.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-base.ts
@@ -18,6 +18,11 @@ import tseslint from 'typescript-eslint';
  * https://github.com/facebook/create-react-app/blob/567f36c9235f1e1fd4a76dc6d1ae00be754ca047/packages/eslint-config-react-app/index.js
  */
 export default tseslint.config({
+  // Files need to be specified or else typescript-eslint rules will be
+  // applied to non-TS files. For example, libs add rules to *.json
+  // files, and TS rules should not apply to them.
+  // See: https://github.com/nrwl/nx/issues/28245
+  files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
   plugins: { import: importPlugin },
   languageOptions: {
     globals: {

--- a/packages/eslint-plugin/src/flat-configs/react-jsx.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-jsx.ts
@@ -15,22 +15,32 @@ import tseslint from 'typescript-eslint';
  */
 export default tseslint.config(
   {
-    // Files need to be specified or else typescript-eslint rules will be
-    // applied to non-TS files. For example, libs add rules to *.json
-    // files, and TS rules should not apply to them.
-    // See: https://github.com/nrwl/nx/issues/28245
-    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
+    files: [
+      '**/*.ts',
+      '**/*.cts',
+      '**/*.mts',
+      '**/*.tsx',
+      '**/*.js',
+      '**/*.cjs',
+      '**/*.mjs',
+      '**/*.jsx',
+    ],
     plugins: {
       'react-hooks': reactHooksPlugin,
     },
     rules: reactHooksPlugin.configs.recommended.rules,
   },
   {
-    // Files need to be specified or else typescript-eslint rules will be
-    // applied to non-TS files. For example, libs add rules to *.json
-    // files, and TS rules should not apply to them.
-    // See: https://github.com/nrwl/nx/issues/28245
-    files: ['**/*.tsx', '**/*.jsx'],
+    files: [
+      '**/*.ts',
+      '**/*.cts',
+      '**/*.mts',
+      '**/*.tsx',
+      '**/*.js',
+      '**/*.cjs',
+      '**/*.mjs',
+      '**/*.jsx',
+    ],
     settings: { react: { version: 'detect' } },
     plugins: {
       'jsx-a11y': jsxA11yPlugin,

--- a/packages/eslint-plugin/src/flat-configs/react-jsx.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-jsx.ts
@@ -15,12 +15,22 @@ import tseslint from 'typescript-eslint';
  */
 export default tseslint.config(
   {
+    // Files need to be specified or else typescript-eslint rules will be
+    // applied to non-TS files. For example, libs add rules to *.json
+    // files, and TS rules should not apply to them.
+    // See: https://github.com/nrwl/nx/issues/28245
+    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
     plugins: {
       'react-hooks': reactHooksPlugin,
     },
     rules: reactHooksPlugin.configs.recommended.rules,
   },
   {
+    // Files need to be specified or else typescript-eslint rules will be
+    // applied to non-TS files. For example, libs add rules to *.json
+    // files, and TS rules should not apply to them.
+    // See: https://github.com/nrwl/nx/issues/28245
+    files: ['**/*.tsx', '**/*.jsx'],
     settings: { react: { version: 'detect' } },
     plugins: {
       'jsx-a11y': jsxA11yPlugin,

--- a/packages/eslint-plugin/src/flat-configs/react-typescript.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-typescript.ts
@@ -11,6 +11,11 @@ import tseslint from 'typescript-eslint';
  * package.
  */
 export default tseslint.config({
+  // Files need to be specified or else typescript-eslint rules will be
+  // applied to non-TS files. For example, libs add rules to *.json
+  // files, and TS rules should not apply to them.
+  // See: https://github.com/nrwl/nx/issues/28245
+  files: ['**/*.ts', '**/*.tsx'],
   rules: {
     // TypeScript"s `noFallthroughCasesInSwitch` option is more robust (#6906)
     'default-case': 'off',

--- a/packages/eslint-plugin/src/flat-configs/react-typescript.ts
+++ b/packages/eslint-plugin/src/flat-configs/react-typescript.ts
@@ -11,11 +11,16 @@ import tseslint from 'typescript-eslint';
  * package.
  */
 export default tseslint.config({
-  // Files need to be specified or else typescript-eslint rules will be
-  // applied to non-TS files. For example, libs add rules to *.json
-  // files, and TS rules should not apply to them.
-  // See: https://github.com/nrwl/nx/issues/28245
-  files: ['**/*.ts', '**/*.tsx'],
+  files: [
+    '**/*.ts',
+    '**/*.cts',
+    '**/*.mts',
+    '**/*.tsx',
+    '**/*.js',
+    '**/*.cjs',
+    '**/*.mjs',
+    '**/*.jsx',
+  ],
   rules: {
     // TypeScript"s `noFallthroughCasesInSwitch` option is more robust (#6906)
     'default-case': 'off',

